mono/utils/mono-log-darwin.c

```diff
@@ -5,7 +5,69 @@
  */
 #include <config.h>
 
-#if defined(HOST_IOS)
+#if defined(HOST_WATCHOS) && (__WATCH_OS_VERSION_MIN_REQUIRED >= __WATCHOS_3_0)
+/* emitted by clang:
+ *   > /Users/lewurm/work/mono-watch4/mono/utils/mono-log-darwin.c:35:2: error: 'asl_log' is \
+ *   > deprecated: first deprecated in watchOS 3.0 - os_log(3) has replaced \
+ *   > asl(3) [-Werror,-Wdeprecated-declarations]
+ */
+
+/* untested stuff: */
+#include <os/log.h>
+#include "mono-logger-internals.h"
+void
+mono_log_open_asl (const char *path, void *userData)
+{
+}
+
+void
+mono_log_write_asl (const char *log_domain, GLogLevelFlags level, mono_bool hdr, const char *message)
+{
```

Comment:
Nit: Assuming `os_log`, `os_log_info`, `os_log_debug`, `os_log_error`, `os_log_fault`  are functions and not macros, refactor to place one of them into a local function pointer and then call `os_log_ptr (OS_LOG_DEFAULT, "%s%s%s\n", log_domain != NULL ? log_domain : "",
				log_domain != NULL ? ": " : "",
				message);`
in just one place, removing about 15 lines of code w/o loss of clarity.

Reply:
they are macros 🙁